src/js/libgeo/GeoOps.js

```diff
@@ -1928,14 +2018,23 @@ geoMacros.Calculation = function(el) {
     return [];
 };
 
+geoMacros.Arc = function(el) {
+    el.type = "ArcBy3";
+    return [el];
+};
+
 geoMacros.Transform = function(el) {
-    var trKind = csgeo.csnames[el.args[0]].kind;
-    var argKind = csgeo.csnames[el.args[1]].kind;
+    var arg = csgeo.csnames[el.args[1]];
+    var tr = csgeo.csnames[el.args[0]];
+    // workaround for Arcs since we treat them as circles
+    var akind = arg.kind;
+    if (arg.kind === "C" && arg.type === "ArcBy3") akind = "Arc";
```

Comment:
Better use `arg.isArc` here, so you can repeatedly apply Möbius transformations to a segment or arc. I think you could drop the `kind === "C"` check and just check the `isArc` property alone.

Perhaps we should consider adding a kind for arcs, if and when we add a kind for circles. These things would be simpler if we had a type hierarchy here, so that we could add subtypes without having to change type checks in a million places. We might want to try the performance impact of such a change one day.